packages/react-components/react-utilities/src/hooks/useAnimationFrame.ts

```diff
@@ -0,0 +1,13 @@
+import { useBrowserTimer } from './useBrowserTimer';
+
+/**
+ * @internal
+ * Helper to manage a browser requestAnimationFrame.
+ * Ensures that the requestAnimationFrame isn't set multiple times at once,
+ * and is cleaned up when the component is unloaded.
+ *
+ * @returns A pair of [requestAnimationFrame, cancelAnimationFrame] that are stable between renders.
+ */
+export function useAnimationFrame() {
+  return useBrowserTimer(requestAnimationFrame, cancelAnimationFrame);
```

Comment:
avoid using document/window globals. use the one from correct window or document object

Reply:
Another PR will address this. The usage of timeout here is the same as we currently have shipped on latest v9 version.